lib/dynflow/director.rb

```diff
@@ -119,7 +119,9 @@ def work_finished(work)
     def work_failed(work)
       if (manager = @execution_plan_managers[work.execution_plan_id])
         manager.terminate
-        finish_manager(manager)
+        # work failed means there was probably a database issue (execution plan gone missing). Don't try
```

Comment:
What are the other cases when this may happen? Can we safely ignore those?

Reply:
In can't think of any other one, but just in case, I added additional check for presence of the execution plan: this should not happen then often, so some additional check to decide whether to store or not make sense.